test/e2e/framework/util.go

```diff
@@ -436,6 +436,13 @@ func WaitForServiceEndpointsNum(ctx context.Context, c clientset.Interface, name
 			return false, nil
 		}
 
+		// Endpoints are single family but EndpointSlices can have dual stack addresses,
+		// so we verify the number of addresses that matches the same family on both.
+		addressType := discoveryv1.AddressTypeIPv4
+		if expectNum > 0 && netutils.IsIPv6String(endpoint.Subsets[0].Addresses[0].IP) {
```

Comment:
can Subsets or Addresses be zero-length and panic this?

Reply:
I can make this more resilient, is not straightforward to notice it

Reply:
done, wait for CI jobs to verify